File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int solution(int n) {
    int answer = 0;
    vector<int> fibo(n+1);
    fibo[0] = 0;
    fibo[1] = 1;
    for(int i=2; i<=n; i++) {
        fibo[i] = (fibo[i-1] + fibo[i-2]) % 1234567;
    }
        
    return fibo[n];
}

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.java
class Solution {
    public int solution(int n) {
        final int MOD = 1234567;
        int[] dp = new int[n+1];
        dp[1] = 1;
        for(int i=2; i<=n; i++) {
            dp[i] = (dp[i-2] + dp[i-1]) % MOD;
        }
        return dp[n] % MOD;
    }
}